src/utils.rs

```diff
@@ -132,6 +136,38 @@ pub(crate) fn write_wallet_from_mnemonic_and_password(
     .map(|_| ())
 }
 
+/// Determine whether to replace an existing wallet
+///
+/// If return true and there is an existing wallet, will be remove first
+/// 
+/// If return false, nothing changes
+pub(crate) fn should_replace_wallet(wallet_path: &Path, force: bool, mut reader: impl BufRead) -> bool {
```

Comment:
Instead of returning a boolean, let's just return `Result<()>`. The error will be printed at the top level when you use `bail!`